BridgeAppSDK/SBAUser.swift

```diff
@@ -41,7 +41,18 @@ import ResearchKit
  */
 public final class SBAUser: NSObject, SBAUserWrapper, SBANameDataSource, SBBAuthManagerDelegateProtocol {
 
-    static let shared = SBAUser()
+    static let shared: SBAUser = {
+        let user = SBAUser()
+        // If a previous install on this device was deleted, its keychain will still be around, but its
+        // userDefaults will have been wiped. In that case we want to make sure the old keychain is wiped
+        // so BridgeSDK doesn't see the old credentials and try to read/write from the old account during
+        // the onboarding/consent/signUp process.
+        if user.onboardingStepIdentifier == nil &&
+            !user.isLoginVerified {
+            user.resetKeychain()
```

Comment:
I think I'd rather put this somewhere it could be overridden. Maybe during appDelegate launch? There are a number of apps that don't use the onboarding step identifier. (Lilly, Smart4SURE, BiAffect)